pystruct/learners/n_slack_ssvm.py

```diff
@@ -12,13 +12,17 @@
 import cvxopt
 import cvxopt.solvers
 
-from sklearn.externals.joblib import Parallel, delayed
+from sklearn.externals.joblib import Parallel, delayed, cpu_count
 from sklearn.utils import gen_even_slices
 
 from .ssvm import BaseSSVM
 from ..utils import unwrap_pairwise, find_constraint
 
 
+def find_constraint_map(args):
+    return find_constraint(* args)
```

Comment:
you can create a new utils file, like utils.parallelization or something. yeah you could also pass a pool. I didn't really think through what would be the best interface.
> Maybe we want a function that takes all the pool options and n_jobs and then decides which parallel thing to use. Or maybe have a utility function that takes a pool and put that in utils, and put a function that does the parallelization as a (private?) method into BaseSSVM.

Reading this more closely I realized there's a few options here. I am currently taking the approach of the utility function in utils.parallel that takes a function, args tuple, pool (that can equal None) and an integer n_jobs (-1, 1 to n_cpus). I was thinking pool allocation could be handled in a private method in BaseSSVM that gets called at the top of fit and inference methods, checks for existence of pool and n_jobs param and creates or passes depending. Also am including a custom pickler there for the model object that strips out pool when dumping and recreates it when loading. that all sound reasonable?

the idea would be to use utility function in all the cases that are equivalently embarrassingly parallel in all the fit methods and the predict method.